### PR TITLE
Ability to launch the importer as a CLI tool 

### PR DESCRIPTION
**Pull request by thomas-hilaire**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    .description("Listen to the message-queue to import data in an OBM instance")
+    .option("-u, --papi_user <user>", "The PAPI user login [admin0@global.virt]", "admin0@global.virt")
+    .option("-p, --papi_pwd <password>", "The PAPI user password [admin]", "admin")
+    .option("-h, --amqp_host <host>", "The AMQP server address [amqp://localhost]", "amqp://localhost")
+    .option("-s, --batch_size <size>", "Maximum item (e.g. event) count in one batch [5]", 5)
+    .option("-w, --batch_wait <wait>", "Time to wait (in ms) for new message from the message-queue before making a batch [5000]", 5000)
+    .option("-d, --batch_delay <delay>", "Minimal time (in ms) to wait between two batches, can be used to avoid PAPI overload [1000]", 1000)
+    .action((papiUrl, domainUuid, options) => {
+
+        let config: ImporterConfig = {
```
Comment (fvn-linagora) [inline src/main.ts]: extract into its own module/object to avoid mixing CLI parsing and papi import ?

Comment (thomas-hilaire) [inline src/main.ts]: Here I only build "config objects" then launch the expected command, the SRP scope small enough IMO. Do you think splitting that again will be an improvement ?

Comment (fvn-linagora) [inline src/main.ts]: An Importer instance creation smells like a candidate for a factory to me.
action's callback is currently newing/setting up an importer and then triggering the import. In my book, that's more than a single resp, isn't it ?

Comment (thomas-hilaire) [inline src/main.ts]: ok I did it

**Comment by fvn-linagora** [comment]
:+1:

**Pull request merged**